ci/lint/lint.sh

```diff
@@ -10,6 +10,14 @@ clang_format() {
   ./ci/lint/check-git-clang-format-output.sh
 }
 
+pre_commit() {
+  # Run pre-commit on all files
+  # TODO(MortalHappiness): Run all pre-commit checks
+  # Currently, we only run the ruff format check
+  pip install -c python/requirements_compiled.txt pre-commit
```

Comment:
`pre-commit` is not a part of `python/requirements_compiled.txt` I think, you might want to pin a version

Reply:
I found that the Python version in Buildkite is 3.8, while the latest version of `pre-commit` requires Python 3.9 or higher. So, I temporarily pinned `pre-commit` to version 3.5.0. Since Ray no longer supports Python 3.8, perhaps the CI team should consider upgrading the Python version in Buildkite to 3.9 or higher. Once that's done, I'll pin `pre-commit` to the latest version.